x/launch/keeper/request.go

```diff
@@ -131,32 +132,62 @@ func CheckAccount(ctx sdk.Context, k Keeper, launchID uint64, address string) (b
 func ApplyRequest(
 	ctx sdk.Context,
 	k Keeper,
-	launchID uint64,
+	chain types.Chain,
 	request types.Request,
+	coord profiletypes.Coordinator,
 ) error {
-	if err := CheckRequest(ctx, k, launchID, request); err != nil {
+	err := CheckRequest(ctx, k, chain.LaunchID, request)
+	if err != nil {
 		return err
 	}
 
 	switch requestContent := request.Content.Content.(type) {
 	case *types.RequestContent_GenesisAccount:
 		ga := requestContent.GenesisAccount
 		k.SetGenesisAccount(ctx, *ga)
+		err = ctx.EventManager().EmitTypedEvent(&types.EventGenesisAccountAdded{
```

Comment:
For all events here, we also emit the event in the message for request when sent by the coordinator. So the event would be sent twice

We should keep only these events here